lib/pages/task_list.dart

```diff
@@ -75,47 +67,62 @@ class _TaskListState extends State<TaskList> {
         context: context,
         builder: (context) {
           return Container(
+            key: const Key('edit_container'),
             padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 60),
             color: bgColor,
             child: EditTask(
                 task: task,
-                onTaskUpdated: (updatedTask) {
-                  context
-                      .read<TaskCubit>()
-                      .updateTask(updatedTask: updatedTask);
+                onTaskUpdated: (String? newName) {
+                  context.read<TaskCubit>().updateTask(task: task, newName: newName);
                 }),
```

Comment:
Nice! I didn't notice this before. You reused the widget and gave it different functionality for both uses. That's well done. Please resolve this comment.